OpenRA.Mods.Cnc/Traits/Buildings/ClonesProducedUnits.cs

```diff
@@ -68,7 +68,7 @@ public void UnitProducedByOther(Actor self, Actor producer, Actor produced, stri
 					factionInit ?? new FactionInit(BuildableInfo.GetInitialFaction(produced.Info, p.Faction))
 				};
 
-				if (p.Produce(self, produced.Info, Info.ProductionType, inits))
+				if (p.Produce(self, new ActorInfo[] { produced.Info }, Info.ProductionType, inits))
```

Comment:
Forcing all these other cases to create transient arrays isn't great for memory churn. Can we please keep the original single-actor `Produce` and define a second method that takes the enumeration? The default virtual implementation on `Production` can then call the single-actor `Produce` in a loop and `BulkProductionQueue` can override it.

Reply:
Made the change less invasive.